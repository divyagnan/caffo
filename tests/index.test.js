const fs = require("fs");
const path = require("path");
const prompts = require("prompts");
const Handlebars = require("handlebars");
const {
  getTemplateDirectoriesAndFiles,
  VariableScanner,
  templateFilePathHelper
} = require("../utils");

test("properly gets template directories and files", () => {
  const testVal = getTemplateDirectoriesAndFiles();

  const expectedVal = {
    dirs: [{ title: "connectedComponent", value: "connectedComponent" }],
    files: [{ title: "reactTemplate.js", value: "reactTemplate.js" }]
  };

  expect(testVal).toEqual(expectedVal);
});

test("properly gets variables from a handlebars template", () => {
  const fakeFile = `
    {{varOne}} {{varTwo}} {{varThree}} 
    {{varOne}} {{varTwo}} {{varThree}}
  `;
  const scanner = new VariableScanner();
  const ast = Handlebars.parse(fakeFile);
  scanner.accept(ast);

  const expectedVal = ["varOne", "varTwo", "varThree"];

  expect([...scanner.variables]).toEqual(expectedVal);
});

// test("e2e", () => {
//   // inject values for testing
//   prompts.inject({
//     templateToUse: "connectedComponent",
//     componentName: "testComponent",
//     lib: "testLib",
//     rootClass: "testClass",
//     directoryToPlaceFiles: "tests/testOutputFolder"
//   });

//   start();

//   setTimeout(() => {
//     // get the files from the expected directory
//     const fileNames = fs.readdirSync(
//       path.join(process.cwd(), "/tests/testOutputFolder")
//     );

//     const fileContents = fileNames.map(fileName => ({
//       name: fileName,
//       content: fs.readFileSync(
//         path.join(process.cwd(), `/tests/testOutputFolder/${fileName}`),
//         "utf8"
//       )
//     }));

//     expect(fileContents).toMatchSnapshot();
//   }, 5000);
// });
