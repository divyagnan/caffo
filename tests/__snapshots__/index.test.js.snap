// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`e2e 1`] = `
Array [
  Object {
    "content": ".testComponent {
    color: red;
}

.testClass {
    color: black;
}",
    "name": "component.css",
  },
  Object {
    "content": "import React from \\"react\\"
import testLib from \\"testLib\\"
import \\"./testComponent.css\\"

class testComponent extends React.Component {
    render() {
        return (
            <div className=\\"testClass\\">Hi from testComponent</div>
        )
    }
}

export default testComponent",
    "name": "component.js",
  },
]
`;
